core/SConscript.bootloader

```diff
@@ -279,7 +279,9 @@ def cargo_build():
         '--no-default-features',
         '--features ' + ','.join(features),
         '-Z build-std=core',
-        '-Z build-std-features=panic_immediate_abort',
+        '-Z build-std=alloc',
+        '-Z build-std=panic_abort,std',
+        # '-Z build-std-features=panic_immediate_abort',
     ]
```

Comment:
Btw, with a relatively recently `nightly`, `make build_bootloader` fails with the error *"building std is not supported on the following targets: thumbv7em-none-eabihf"* (if building for Model T, for other models the platform may be different).
So I tried `nightly-2024-01-21`, which I believe they use on CI. This worked, but I also had to lower the rust version for `trezor-common` to 1.77. The question is - should we also always use a specific version of `nightly`? It looks like we should, otherwise things may break unexpectedly. But `trezor-common` will have to use a lower-than-normal rust version.
I also wonder what version of `nightly` you used.

P.S.
1. `build-std=...,std` means that it's no longer no_std, which is not good.
2. only the last `-Z build-std` option seems to apply. So you probably only need `-Z build-std=core,alloc`, same for `core/SConscript.firmware`.
3. We probably shouldn't require `alloc` for the bootloader. E.g., we may make `parity-scale-codec` and `ml_common` optional in `Cargo.toml` (`optional = true`) and then add a feature `mintlayer = ["dep:parity-scale-codec", "dep:ml_common"]` and then make it a dependency for the `universal_fw` feature: `universal_fw = ["mintlayer"]`. (just tried it, seems to work)

Reply:
removed std, forgot to do that before